src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -14,17 +20,335 @@ public abstract class TextEncoder
         // and so I think the warning is a bit too aggressive.  
         #pragma warning disable 3011
         [CLSCompliant(false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public unsafe abstract bool TryEncodeUnicodeScalar(int unicodeScalar, char* buffer, int bufferLength, out int numberOfCharactersWritten);
 
         // all subclasses have the same implementation of this method.
         // but this cannot be made virtual, because it will cause a virtual call to Encodes, and it destroys perf, i.e. makes common scenario 2x slower 
         [CLSCompliant(false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public unsafe abstract int FindFirstCharacterToEncode(char* text, int textLength);
         #pragma warning restore
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public abstract bool Encodes(int unicodeScalar);
 
         // this could be a field, but I am trying to make the abstraction pure.
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public abstract int MaxOutputCharactersPerInputCharacter { get; }
+
+        public string Encode(string value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException("value");
+            }
+
+            unsafe
+            {
+                fixed (char* valuePointer = value)
+                {
+                    int firstCharacterToEncode = FindFirstCharacterToEncode(valuePointer, value.Length);
+
+                    if (firstCharacterToEncode == -1)
+                    {
+                        return value;
+                    }
+
+                    int bufferSize = MaxOutputCharactersPerInputCharacter * value.Length;
+                    char* wholebuffer = stackalloc char[bufferSize];
```

Comment:
I don't off the top of my head know any number, but you could try to run an analysis on something like the MusicStore app to record the lengths to get *some* idea. However, MusicStore is a relatively simple app so that's just one datapoint. So I would use MusicStore as a "minimum" exemplar (as in, all apps would do *at least* what MusicStore does).